client/src/commands/authorize.ts

```diff
@@ -0,0 +1,23 @@
+import { authentication, commands } from "vscode";
+import { SASAuthProvider } from "../components/AuthProvider";
+import { profileConfig, switchProfile } from "./profile";
+
+export const checkProfileAndAuthorize = async (): Promise<void> => {
+  commands.executeCommand("setContext", "SAS.authorizing", true);
+  if (profileConfig.getActiveProfile() === "") {
+    await switchProfile();
+  }
+
+  if (profileConfig.getActiveProfile() !== "") {
+    try {
+      await authentication.getSession(SASAuthProvider.id, [], {
+        createIfNone: true,
+      });
+    } catch (error) {
+      // Do nothing here. We just want to make sure we're resetting
```

Comment:
I'm thinking we should pop any errors up to let user know. And to fail any further actions.
resetting context can be in a "finally" block.

Reply:
I've added a call to `window.showErrorMessage`. Although, is pressing escape to prevent auth code entry _really_ an error (in that case, the user _knows_ what they did)?

Reply:
But there can be other errors like network or server reported errors.
We did see a few errors during authorize when making run code, so made below for error reporting.
https://github.com/sassoftware/vscode-sas-extension/blob/main/client/src/commands/run.ts#L105